dhis2/src/main/java/org/motechproject/dhis2/service/Settings.java

```diff
@@ -14,7 +14,11 @@
     public Settings() { }
 
     public Settings(String serverURI, String username, String password) {
-        this.serverURI = serverURI;
+        if (serverURI.endsWith("/")) {
```

Comment:
Including this logic at the construct level is not the best idea. If I use setter instead of the constructor to provide the server URL this won't be invoked and thus manifest itself again.